logging/src/main/java/org/jboss/as/logging/filters/FilterResourceDefinition.java

```diff
@@ -0,0 +1,250 @@
+ */
+public class FilterResourceDefinition extends TransformerResourceDefinition {
+    public static final String NAME = "filter";
+
+    public static final SimpleMapAttributeDefinition CONSTRUCTOR_PROPERTIES = new SimpleMapAttributeDefinition.Builder("constructor-properties", true)
+            .setAllowExpression(true)
+            .setAttributeMarshaller(PropertyAttributeMarshaller.INSTANCE)
+            .setXmlName("constructor-properties")
+            .build();
+
+    private static final PathElement PATH = PathElement.pathElement(NAME);
+
+    private static final AttributeDefinition[] ATTRIBUTES = {
+            CLASS,
+            MODULE,
+            CONSTRUCTOR_PROPERTIES,
+            PROPERTIES,
+    };
+
+
+    /**
+     * A step handler to add a custom filter
+     */
+    private static final OperationStepHandler ADD = new LoggingOperations.LoggingAddOperationStepHandler(ATTRIBUTES) {
+        private final List<String> reservedNames = Arrays.asList(
+                "accept",
+                "deny",
+                "not",
+                "all",
+                "any",
+                "levelChange",
+                "levels",
+                "levelRange",
+                "match",
+                "substitute",
+                "substituteAll"
+        );
+
+        @Override
+        protected void populateModel(final OperationContext context, final ModelNode operation, final Resource resource) throws OperationFailedException {
+            // Check the name isn't a reserved filter name
+            final String name = context.getCurrentAddressValue();
+            if (reservedNames.contains(name)) {
+                throw LoggingLogger.ROOT_LOGGER.reservedFilterName(name, reservedNames);
+            }
+            // Check the name has no special characters
+            if (!Character.isJavaIdentifierStart(name.charAt(0))) {
+                throw LoggingLogger.ROOT_LOGGER.invalidFilterNameStart(name, name.charAt(0));
+            }
+            for (char c : name.toCharArray()) {
+                if (!Character.isJavaIdentifierPart(c)) {
+                    throw LoggingLogger.ROOT_LOGGER.invalidFilterName(name, c);
+                }
+            }
+            super.populateModel(context, operation, resource);
+        }
+
+        @Override
+        public void performRuntime(final OperationContext context, final ModelNode operation, final ModelNode model, final LogContextConfiguration logContextConfiguration) throws OperationFailedException {
+            final String name = context.getCurrentAddressValue();
+            FilterConfiguration configuration = logContextConfiguration.getFilterConfiguration(name);
+            final String className = CLASS.resolveModelAttribute(context, model).asString();
+            final ModelNode moduleNameNode = MODULE.resolveModelAttribute(context, model);
+            final String moduleName = moduleNameNode.isDefined() ? moduleNameNode.asString() : null;
```

Comment:
No need to change this but in the future you can do

final String moduleName = MODULE.resolveModelAttribute(context, model).asStringOrNull();

Reply:
I saw that on a PR the other day and was quite surprised I didn't know about it :)